src/tainting/Dataflow_tainting.ml

```diff
@@ -1727,7 +1727,14 @@ let findings_from_arg_updates_at_exit enter_env exit_env : T.finding list =
   (* TOOD: We need to get a map of `lval` to `Taint.arg`, and if an extension
    * of `lval` has new taints, then we can compute its correspoding `Taint.arg`
    * extension and generate an `ArgToArg` finding too. *)
-  enter_env |> Lval_env.seq_of_tainted |> List.of_seq
+  enter_env |> Lval_env.seq_of_tainted
+  |> Seq.flat_map (fun (lval, enter_taints) ->
+         (* We need to consider all lvals of the same base component
```

Comment:
So, `x` is an input "arg" and `x[1]` gets tainted. Cannot this happen also with fields? E.g. a class has a field `x` and we taint `x.a`. This will be more natural with "taint shapes", but fine to have this fix if it's blocking Flask decorators.

Reply:
This PR covers both cases: https://github.com/semgrep/semgrep-proprietary/pull/1333